### ISSUE TITLE
Why there is a type check before calling postMessage?

### ISSUE THREAD

**Issue by cagriyildiz** [SonarSource/SonarJS#2052]
https://github.com/SonarSource/SonarJS/blob/bfb30ef97a6795ee0a02d3ecd84f17a0df526974/javascript-checks/src/main/java/org/sonar/javascript/checks/PostMessageCheck.java#L43

What is the reason of this type check? Will I get faced this issue every time I use this function because of this type check?
`callee.object().types().contains(Type.Kind.WINDOW)`

postMessage is called only from the window object. https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage
Every time postMessage function is called this rule will occur whether you use window.postMessage or use a variable before calling postMessage like this:
```
var content = this.elem.contentWindow;
content.postMessage('your message', window.location.origin);
```

This answer on stackoverflow says that this approach bypass this issue but not worked for me.
https://stackoverflow.com/questions/54139298/sonarjs-still-shows-warning-about-postmessage-cross-domain-issue

**Comment by vilchik-elena** [comment]
Hi,

Could you describe your initial problem? Do you have some kind of false positive?

**Comment by cagriyildiz** [comment]
I think it is false positive.

My initial code was:

**Comment by vilchik-elena** [comment]
As you can see in #1772 we plan to report only when `*` is used as parameter. So considering your new code (second code sample), it seems to be safe and I would just close the issue as false-positive.
